src/app/components/store-products/store-products.component.ts

```diff
@@ -1,10 +1,52 @@
-import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, combineLatest } from 'rxjs';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { StoreDetailsQueryModel } from '../../query-models/store-details.query-model';
+import { StoreModel } from '../../models/store.model';
+import { ProductModel } from '../../models/product.model';
+import { StoresService } from '../../services/stores.service';
+import { ProductsService } from '../../services/products.service';
 
 @Component({
   selector: 'app-store-products',
   templateUrl: './store-products.component.html',
   encapsulation: ViewEncapsulation.None,
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class StoreProductsComponent {
+  readonly storeDetails$: Observable<StoreDetailsQueryModel> = combineLatest([
+    this._activatedRoute.params.pipe(
+      switchMap((params) => this._storesService.getOneStore(params['storeId']))
+    ),
+    this._productsService.getAllProducts(),
+  ]).pipe(
+    map(([store, products]: [StoreModel, ProductModel[]]) => {
+      const productsByStore: ProductModel[] = products.filter((product) =>
+        product.storeIds.includes(store.id)
+      );
```

Comment:
I have filter and then map - could I use reduce here? I tried to create productsMap but I did'nt know how to extract a single storeId as a key + storeId is not a unique value - several products can have the same storeId

Reply:
You can try something like this:

![image](https://user-images.githubusercontent.com/82345363/215087495-e1ff5c82-c23e-4e23-bd57-d52e8e5fde37.png)

Reply:
You can, In the reduce method you can check this condition include and when its true, so return the array expanded by the current value. You can expand the current array with the mapped object. So reduce method should return array with correct data.